src/commands/build/watch/mod.rs

```diff
@@ -54,7 +59,27 @@ pub fn watch_and_build(
                 let (watcher_tx, watcher_rx) = mpsc::channel();
                 let mut watcher = notify::watcher(watcher_tx, Duration::from_secs(1)).unwrap();
 
-                watcher.watch(RUST_PATH, RecursiveMode::Recursive).unwrap();
+                // Populate walker with ignored files so we ensure that the watcher does not watch
+                // ignored directories
+                let mut ignored_files = OverrideBuilder::new("./");
+                for ignore in RUST_IGNORE {
+                    ignored_files.add(&format!("!{}", ignore)).unwrap();
+                }
+                let ignored_file_override = ignored_files.build().unwrap();
+
+                let walker = WalkBuilder::new("./")
+                    .overrides(ignored_file_override)
+                    .build();
+
+                for entry in walker {
+                    let entry = entry.unwrap();
+                    if entry.path().is_dir() {
+                        continue;
```

Comment:
what is the purpose of this? does `continue` jump it back up to the start of the loop?

Reply:
we don't want to watch over entire directories; this will allow for unwanted files to be watched (and trigger continuous loading). we skip and continue looping.